src/software/proto/message_translation/ssl_message_translator.h

```diff
@@ -0,0 +1,180 @@
+          LEFT_FIELD_LEFT_PENALTY_STRETCH,
+          LEFT_FIELD_RIGHT_PENALTY_STRETCH,  // right is positive y
+          // For the RIGHT_FIELD, LEFT and RIGHT are from the POV of the net
+          // on the right of the field looking in towards the center of the
+          // field. ie. LEFT is +y, RIGHT is -y
+          RIGHT_FIELD_LEFT_PENALTY_STRETCH, RIGHT_FIELD_RIGHT_PENALTY_STRETCH, )
+
+static const std::map<SSLFieldLines, const std::string> ssl_field_line_names = {
+    {SSLFieldLines::TOP_TOUCH_LINE, "TopTouchLine"},
+    {SSLFieldLines::BOTTOM_TOUCH_LINE, "BottomTouchLine"},
+    {SSLFieldLines::LEFT_GOAL_LINE, "LeftGoalLine"},
+    {SSLFieldLines::RIGHT_GOAL_LINE, "RightGoalLine"},
+    {SSLFieldLines::HALFWAY_LINE, "HalfwayLine"},
+    {SSLFieldLines::CENTER_LINE, "CenterLine"},
+    {SSLFieldLines::LEFT_PENALTY_STRETCH, "LeftPenaltyStretch"},
+    {SSLFieldLines::RIGHT_PENALTY_STRETCH, "RightPenaltyStretch"},
+    {SSLFieldLines::RIGHT_GOAL_TOP_LINE, "RightGoalTopLine"},
+    {SSLFieldLines::RIGHT_GOAL_BOTTOM_LINE, "RightGoalBottomLine"},
+    {SSLFieldLines::RIGHT_GOAL_DEPTH_LINE, "RightGoalDepthLine"},
+    {SSLFieldLines::LEFT_GOAL_TOP_LINE, "LeftGoalTopLine"},
+    {SSLFieldLines::LEFT_GOAL_BOTTOM_LINE, "LeftGoalBottomLine"},
+    {SSLFieldLines::LEFT_GOAL_DEPTH_LINE, "LeftGoalDepthLine"},
+    {SSLFieldLines::LEFT_FIELD_LEFT_PENALTY_STRETCH, "LeftFieldLeftPenaltyStretch"},
+    {SSLFieldLines::LEFT_FIELD_RIGHT_PENALTY_STRETCH, "LeftFieldRightPenaltyStretch"},
+    {SSLFieldLines::RIGHT_FIELD_LEFT_PENALTY_STRETCH, "RightFieldLeftPenaltyStretch"},
+    {SSLFieldLines::RIGHT_FIELD_RIGHT_PENALTY_STRETCH, "RightFieldRightPenaltyStretch"},
+};
+
+MAKE_ENUM(SSLCircularArcs, CENTER_CIRCLE)
+
+static const std::map<SSLCircularArcs, const std::string> ssl_circular_arc_names = {
+    {SSLCircularArcs::CENTER_CIRCLE, "CenterCircle"},
+};
+
+/**
+ * Find an SSL FieldLineSegment by name. Returns the first instance of a line segment
+ * with the given name if it exists, otherwise returns std::nullopt
+ *
+ * @param line_segments The line segments to search
+ * @param name The name of the segment to find
+ *
+ * @return The first line segment with the given name if it exists, otherwise
+ * returns std::nullopt
+ */
+std::optional<SSL_FieldLineSegment> findLineSegment(
+    const google::protobuf::RepeatedPtrField<SSL_FieldLineSegment>& line_segments,
+    const std::string& name);
+
+/**
+ * Find an SSL FieldCircularArc by name. Returns the first instance of a field arc
+ * with the given name if it exists, otherwise returns std::nullopt
+ *
+ * @param circular_arcs The arcs to search
+ * @param name The name of the arc to find
+ *
+ * @return The first arc with the given name if it exists, otherwise
+ * returns std::nullopt
+ */
+std::optional<SSL_FieldCircularArc> findCircularArc(
+    const google::protobuf::RepeatedPtrField<SSL_FieldCircularArc>& circular_arcs,
+    const std::string& name);
```

Comment:
Could we put the maps in the `.cpp` file and then just pass Enums into this? That we we totally encapsulate the fact that SSL uses raw strings in this way.

Reply:
done